internal/app/api/handler/user/user.go

```diff
@@ -98,13 +115,17 @@ func (h *Handler) UpdateUserByID(w http.ResponseWriter, r *http.Request) {
 	err := json.NewDecoder(r.Body).Decode(&user)
 
 	if err != nil {
-		respond.JSON(w, http.StatusInternalServerError, h.em.InvalidValue.Request)
+		respond.JSON(w, http.StatusBadRequest, h.em.InvalidValue.Request)
```

Comment:
use if err:= ...; err {} incase no new variable is created
Log error before response